python/lsst/pipe/tasks/mocks/simpleMapper.py

```diff
@@ -103,6 +103,34 @@ def makeButlerLocation(cls, path, dataId, mapper, suffix=None):
                                        dataId['imageOrigin'])
         return loc
 
+class ImageDPersistenceType(PersistenceType):
+    """Persistence type of ImageD images.
+    """
+
+    python = "lsst.afw.image.ImageD"
+    cpp = "ImageD"
+    storage = "FitsStorage"
+    ext = ".fits"
+    suffixes = ("_sub",)
+
+    @classmethod
+    def makeButlerLocation(cls, path, dataId, mapper, suffix=None):
+        """Method called by SimpleMapping to implement a map_ method; overridden to support subimages."""
```

Comment:
Do you need all this stuff about supporting subimages? Maybe it's necessary; it's not easy for me to tell without more thought. Even if it's not necessary, maybe it's easier just to copy the exposure persistence than to try and disentangle it.

That said `ExposurePersistenceType` and `ImageDPersistenceType` are, I think, almost identical: you could probably save a lot of typing (or copying & pasting!) by moving `makeButlerLocation` to a common base.